src/pages/CohortManagerMainPage.jsx

```diff
@@ -0,0 +1,139 @@
+      document.getElementById("submit-button").disabled = "disabled";
+    } else {
+      console.log("enable button");
+      document.getElementById("submit-button").removeAttribute("disabled");
+    }
+  }, [newPostData]);
+
+  const handleInputPostData = (event) => {
+    const postTextValue = event.target.value;
+
+    setNewPostData(postTextValue);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const postRequestData = stringToPostRequestBody(newPostData);
+
+    MakeAPIPostRequest(postRequestData);
+
+    setNewPostData(InitialPostData); // this resets the text box
+  };
+
+  //Splits the input text to the title and content and puts them in a postRequestBody object
+  const stringToPostRequestBody = (newPostData) => {
+    const splitTitleAndText = newPostData.split("|");
+    const postRequestBody = {
+      title: splitTitleAndText[0],
+      content: splitTitleAndText[1],
+      contactId: 1,
+    };
+    return postRequestBody;
+  };
+
+  const MakeAPIPostRequest = (postRequestData) => {
+    const postRequestOption = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(postRequestData),
+    };
+
+    makeNewPostToAPI(postRequestOption, setPostsData, postsData);
+
+    console.log(postRequestOption);
+  };
+
+  return (
+    <div className="container">
+      <header className="header header-colour">
+        <MainPageIcon />
+      </header>
+      <div className="container-nav-main">
+        <nav className="sidebar sidebar-colour">
+          <button className="sidebar-button">
+            <HomePageIcon />
+            <p className="sidebar-button-image-and-font">Home</p>
+          </button>
+          <button className="sidebar-button">
+            <ProfileIcon />
+            <p>Profile</p>
+          </button>
+        </nav>
+        <main className="main background">
+          <div className="post-bar-box">
+            <div className="circle">
+              {simulatedUserData && (
+                <p className="text">
+                  {GetInitalsFromNames(
+                    simulatedUserData.firstName,
+                    simulatedUserData.lastName
+                  )}
+                </p>
+              )}
+            </div>
+            <form onSubmit={handleSubmit}>
+              <input
+                className="post-bar-text-field"
+                placeholder="Write your post here. To separate post title and post text you ned to add a: | between them."
+                type="text"
+                name="postText"
+                value={newPostData}
+                onChange={handleInputPostData}
+              />
+              <button
+                id="submit-button"
+                className="post-bar-button"
+                disabled="disabled"
+              >
+                Post
+              </button>
+            </form>
+          </div>
+          {postsData.toReversed().map((postData, index) => (
```

Comment:
Did you mean to use 'reverse()' here? Be cautious as reverse() mutates the array. To avoid mutating postsData directly, you can create a shallow copy before reversing

Reply:
No, I intended to use toReversed() here. It creates a copy of postData and reverses it. Here is where I read about it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toReversed#Overview